jobbergate-api/jobbergate_api/version.py

```diff
@@ -0,0 +1,43 @@
+"""
+Provide the version of the package.
+"""
+
+from importlib import metadata
+
+import toml
+
+
+def get_version_from_metadata() -> str:
+    """
+    Get the version from the metadata.
+
+    This is the preferred method of getting the version, but only works if the
+    package is properly installed in a Python environment.
+    """
+    return metadata.version(__package__ or __name__)
+
+
+def get_version_from_poetry() -> str:
+    """
+    Get the version from pyproject.toml.
+
+    This is a fallback method if the package is not installed, but just copied,
+    like in a Dockerfile.
+    """
+    return toml.load("pyproject.toml")["tool"]["poetry"]["version"]
+
+
+def get_version() -> str:
+    """
+    Get the version from the metadata if available, otherwise from pyproject.toml.
+    """
+    try:
+        return get_version_from_metadata()
+    except metadata.PackageNotFoundError:
+        try:
+            return get_version_from_poetry()
+        except (FileNotFoundError, KeyError):
+            return "0.0.0"
```

Comment:
I'm also not a big fan of this complexity, but `get_version_from_metadata` is the way to go for any user that has installed Jobbergate-API from PyPI or from the source. On the other hand, `get_version_from_poetry` is there due to the logic on the docker containers we have for internal use, they just copy the API code instead of installing it.

We can fix the Dockerfiles and simplify things here, but for me, that is out of the scope of this task.

About version `0.0.0`, you have a good point, I'll check if `None` works there to avoid any confusion.

Reply:
FastAPI checks if the version is a non-empty string, so the fallback value was changed to `"unknown"` in [6b0910b](https://github.com/omnivector-solutions/jobbergate/pull/185/commits/6b0910b0e8adaff56e7625a303b8daafa19079a1).